include/ipc/ipc_icmsg_buf.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_IPC_SERVICE_IPC_ICMSG_BUF_H_
+#define ZEPHYR_INCLUDE_IPC_SERVICE_IPC_ICMSG_BUF_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief IPC Service ICMsg buffer API
+ * @ingroup ipc_service_icmsg_buffer_api IPC service ICMsg buffer API
+ * @{
+ */
+
+/**
+ * @brief Inter core messaging buffer
+ *
+ * The inter core messaging buffer implements lightweight unidirectional
+ * messaging buffer with read/write semantics on top of a memory region shared
+ * by the reader and writer. It embeds cache and memory barier management to
+ * ensure correct data access.
+ *
+ * This structure supports single writter and reader. Data stored in the buffer
+ * is encapsulated to a message.
+ *
+ */
+struct icmsg_buf {
+	uint32_t len;		/* Length of data[] in bytes. */
```

Comment:
I don't see any use of those flags for now. If anyone would later on expand this - i have no objections. For now this is minimal required version and i would like to avoid adding something that _could be_ useful in the future.